server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -80,12 +79,25 @@ private static boolean searchWithCollectorManager(
         try {
             final ReduceableSearchResult result = searcher.search(query, collectorManager);
             result.reduce(queryResult);
-        } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
-            queryResult.terminatedEarly(true);
-        } catch (TimeExceededException e) {
+        } catch (RuntimeException re) {
```

Comment:
Shouldn't we keep the `EarlyTerminationException` as well? When there is only 1 slice, the search would happen in the current thread

Reply:
While the search will happen in the search thread, the `forceTermination` flag in `EarlyTerminatingCollectorManager` would still always be false: https://github.com/opensearch-project/OpenSearch/blob/1118dcf3727b6a2dee598159d65560adfc4d0dec/server/src/main/java/org/opensearch/search/query/EarlyTerminatingCollectorManager.java#L39-L42

When the flag is false, then we will never return `EarlyTerminationException`, only `CollectionTerminatedException`
https://github.com/opensearch-project/OpenSearch/blob/1118dcf3727b6a2dee598159d65560adfc4d0dec/server/src/main/java/org/opensearch/search/query/EarlyTerminatingCollector.java#L74-L98

And `CollectionTerminatedException` is already handled in `ContextIndexSearcher::searchLeaf` like so https://github.com/opensearch-project/OpenSearch/blob/9b5fa6bb8308a412bf439b7ef97eaf8d68437e80/server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java#L314-L323

Reply:
> When the flag is false, then we will never return EarlyTerminationException, only CollectionTerminatedException

That's I think we will change at some point (the issue you have mentioned), and then very likely we would need to bring it back?